src/zkevm_specs/state.py

```diff
@@ -0,0 +1,544 @@
+from .util import FQ, RLC, U160, U256
+from typing import NamedTuple, Tuple, List, Sequence
+from enum import IntEnum, auto
+from .encoding import U8, is_circuit_code
+from .evm import RW, RWTableTag
+from .evm import AccountFieldTag, CallContextFieldTag
+
+MAX_KEY_DIFF = 2**32 - 1
+MAX_MEMORY_ADDRESS = 2**32 - 1
+MAX_STACK_PTR = 1023
+
+
+class Tag(IntEnum):
+    """
+    Tag used as first key in the State Circuit Rows to "select" the operation target.
+    """
+
+    Start = 1
```

Comment:
Is `Start` aim for avoiding the "previous row wraps around" issue? It seems we can leverage this as padding (prover adds as many `Start` rows in the beginning with `rw_counter = 0` as padding).

Reply:
> Is `Start` aim for avoiding the "previous row wraps around" issue? 

Yes exactly!

> It seems we can leverage this as padding (prover adds as many `Start` rows in the beginning with `rw_counter = 0` as padding).

That's an interesting idea!  So we pad the beginning rows with Start as necessary, and also use Start to not check rules that would fail due to the wrap around with the end of the table.